hngprojects/hng_boilerplate_golang_web
# Feat: help center

​
## Description

​Added endpoints to the Help Center module for managing help center topics. This includes functionality to create, fetch, search, update and delete help center topics, allowing for efficient management and retrieval of help center content. The author field is automatically set to 'ADMIN'.

[API DOC LINK](https://staging.api-golang.boilerplate.hng.tech/api/docs/index.html)
## Endpoints

- POST /api/v1/help-center/topics - Create a new help center topic
- GET /api/v1/help-center/topics - Retrieve all help center topics
- GET /api/v1/help-center/topics/:id - Retrieve a help center topic by ID
- GET /api/v1/help-center/topics/search - Search help center topics by criteria
- PATCH /api/v1/help-center/topics/:id - Update a help center topic by ID
- DELETE /api/v1/help-center/topics/:id - Delete a help center topic by ID

## Related Issue (Link to Github issue)




[#270​](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/270)
[#271](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/271)
[#272](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/272)
[#273](https://github.com/hngprojects/hng_boilerplate_golang_web/issues/273)
[#76](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/76)

## How Has This Been Tested?



​1. Create a New Help Center Topic

Action: Send a POST request to the POST /api/v1/help-center/topics endpoint with a valid CreateHelpCenter payload.
Expected Result: You should receive a response with status code 201 and the created help center topic object.
2. Retrieve All Help Center Topics

Action: Send a GET request to the GET /api/v1/help-center/topics endpoint.
Expected Result: You should receive a response with status code 200 and an array of help center topics.
3. Retrieve Help Center Topic by ID

Action: Send a GET request to the GET /api/v1/help-center/topics/:id endpoint with a valid topic ID.
Expected Result: You should receive a response with status code 200 and the help center topic with the specified ID
4. Search Help Center Topics

Action: Send a GET request to the GET /api/v1/help-center/topics/search?title={title} endpoint with search criteria in the query parameters.
Expected Result: You should receive a response with status code 200 and an array of help center topics that match the search criteria.
5. Update Help Center Topics

Action: Send a PATCH request to the PATCH /api/v1/help-center/topics/:id endpoint with the updated field in the request body.
Expected Result: You should receive a response with status code 200 and an updated help center topic that match the updated field.
6. Delete Help Center Topics

Action: Send a DELETE request to the DELETE /api/v1/help-center/topics/:id endpoint.
Expected Result: You should receive a response with status code 204 No content.

## Screenshots (if appropriate - Postman, etc):
​
![Screenshot (336)](https://github.com/user-attachments/assets/a3c6f7fd-7f81-4181-9ce3-3f424e4e4ab9)
![Screenshot (337)](https://github.com/user-attachments/assets/e2f19458-a321-44bf-a0bc-434777eef51b)
![Screenshot (338)](https://github.com/user-attachments/assets/ba590052-3cd1-4ffa-8f05-26e5b1aaefc3)
![Screenshot (339)](https://github.com/user-attachments/assets/e364bb98-9fae-41d9-b026-ea7c01f4c09d)
![Screenshot (340)](https://github.com/user-attachments/assets/cdc6148b-35f1-47e0-b516-0d4d5317bdbe)
![Screenshot (341)](https://github.com/user-attachments/assets/954dacac-ee06-4b78-bbaf-c2c0cee0347c)
![Screenshot (342)](https://github.com/user-attachments/assets/87884deb-7495-48cf-bf7a-b6805a012aac)

## Types of changes

​
## Checklist:
